### PR TITLE
Wandb self contained offline logs

### PR DESCRIPTION
**Pull request by le1nux**
# What does this PR do?

This PR makes sure that weights and biases logs everything (i.e., caching, logs etc.) within a single directory and respective subfolders. 

## General Changes
* see above

## Breaking Changes
* none

## Checklist before submitting final PR
- [x] My PR is minimal and addresses one issue in isolation
- [x] I have merged the latest version of the target branch into this feature branch
- [x] I have reviewed my own code w.r.t. correct implementation, missing type hints, proper documentation, etc.
- [x] I have run a sample config for model training
- [x] I have checked that all tests run through (`python tests/tests.py`)

### CODE REVIEW

**Review by fromm-m — APPROVED** [review]

**Review by fromm-m — APPROVED** [review]

**Comment by fromm-m** [comment]
After the tests are fixed it can be merged!

File: config_files/training/config_lorem_ipsum.yaml [hunk 1]
```diff
@@ -306,7 +306,7 @@ evaluation_subscriber:
   config:
     global_rank: ${settings.cuda_env.global_rank}
     project: modalities_lorem_ipsum
-    mode: ONLINE
+    mode: OFFLINE
     experiment_id: ${settings.experiment_id}
-    directory: "."
+    directory: "wandb_storage"
```
Comment (flxst) [inline config_files/training/config_lorem_ipsum.yaml:311]: For the sake of consistency, we should perhaps apply the same change to all the other config files where it is relevant, e.g. 
* `config_files/training/*.yaml`
* `examples/getting_started/example_config.yaml`
* `examples/library_usage/config_lorem_ipsum.yaml`
* [..]

File: src/modalities/logging_broker/subscriber_impl/subscriber_factory.py [hunk 2]
```diff
@@ -67,8 +68,25 @@ def get_wandb_result_subscriber(
...
+                os.environ["WANDB_DATA_DIR"] = str(absolute_dir)
+                os.environ["WANDB_ARTIFACT_LOCATION"] = str(absolute_dir)
+                os.environ["WANDB_ARTIFACT_DIR"] = str(absolute_dir)
+                os.environ["WANDB_CONFIG_DIR"] = str(absolute_dir)
+            else:
+                absolute_dir = None
+
             result_subscriber = WandBEvaluationResultSubscriber(
-                project, experiment_id, mode, directory, config_file_path
+                project, experiment_id, mode, absolute_dir, config_file_path
```
Comment (flxst) [inline src/modalities/logging_broker/subscriber_impl/subscriber_factory.py:89]: Minor typing issue not introduced by this PR: The typing for `absolute_dir` / `logging_directory` in `WandBEvaluationResultSubscriber` is `Path`, but it should be `Optional[Path]`.

**Review by flxst — APPROVED** [review]
LGTM! Left 2 minor comments.

**Pull request merged**